### PR TITLE
feat: OpenAPI 3.1 support (parsers, generators, comprehensive tests)

### LINKED ISSUE #79: OpenAPI 3.1 support
Thank you for this great project!

Would be awesome if OpenAPI 3.1 was supported in addition to OpenAPI 3.0.

It looks like [openapi-pydantic](https://github.com/MarcoMuellner/openapi-python-generator/blob/90643ec0b7a40e53ed39b38628c0d0f24a8b378c/src/openapi_python_generator/generate_data.py#L14) is being used for the OpenAPI parsing, and that project has support for 3.1.

The OpenAPI schema I'm working with comes from [Pydantic JSON schema](https://docs.pydantic.dev/latest/concepts/json_schema/), which is complying to version 3.1 but not 3.0.

### PR DESCRIPTION
**Pull request by dougborg**
Closes #79

## Summary
Adds end-to-end OpenAPI 3.1 support: version detection, dedicated 3.0 & 3.1 parsers, unified model & service code generation, and extensive test coverage.

## Key Changes
- Version detection utility + separate OpenAPI 3.0 / 3.1 parser modules
- Model generator enhancements: refs, composite schemas (allOf / oneOf / anyOf), DataType enums, self-references, import aggregation fix for allOf
- Service generator updates: multi-version media/response handling and request body resolution
- Generalized `generator` & `api_config` for 3.0+ (model_dump usage, shared unions)
- Unified core model unions (Operation, PathItem, Schema) enabling cross-version typing
- New comprehensive test suites:
  - Completeness & schema feature coverage for 3.1
  - Negative spec tests (validation & unsupported features)
  - Swagger Petstore 3.0 & 3.1 regression fixtures
  - Edge cases (self-references, composites, imports)
- Tooling: dependency and typing improvements, test fixture additions

## Test Status
`poetry run pytest -q` → 232 passed, 11 xfailed (intentional skips for unsupported 3.1 features not yet implemented)

## Follow-ups (separate PRs)
- Implement currently xfailed 3.1 features (e.g. unevaluatedProperties, advanced boolean schema behaviors)
- Potential type conversion refactor (see #98)

## Backward Compatibility
No breaking changes for OpenAPI 3.0 users; 3.1 is auto-detected and handled transparently.

Feedback welcome.

### CODE REVIEW

File: src/openapi_python_generator/language_converters/python/service_generator.py [hunk 1]
```diff
@@ -24,6 +46,39 @@
...
+    return isinstance(obj, (Response30, Response31))
+
+
+def create_media_type_for_reference(reference_obj):
+    """Create a MediaType wrapper for a reference object, using the correct version"""
+    # Check which version the reference object belongs to
+    if isinstance(reference_obj, Reference30):
+        return MediaType30(schema=reference_obj)
+    elif isinstance(reference_obj, Reference31):
+        return MediaType31(schema=reference_obj)
```
Comment (MarcoMuellner) [inline src/openapi_python_generator/language_converters/python/service_generator.py:61]: Similar for these types. 

I am also not a 100% sure that this type of abstraction is ... good? It feels a bit icky to do it this way. Rather, i'd have a new type `MediaType` that within itself does the abstraction for it if you understand what i mean.
